workers/unity/Packages/com.improbable.gdk.tools/DownloadCoreSdk.cs

```diff
@@ -136,6 +175,129 @@ private static DownloadResult Download()
             return exitCode == 0 ? DownloadResult.Success : DownloadResult.Error;
         }
 
+        /// <summary>
+        ///     Sets plugin platform compatibility based on directory structure
+        /// </summary>
+        internal static void SetPluginsCompatibility()
+        {
+            AssetDatabase.StartAssetEditing();
+            foreach (var pluginDirectoryCompatibility in PluginsCompatibilityList)
+            {
+                if (!Directory.Exists(pluginDirectoryCompatibility.Path))
+                {
+                    continue;
+                }
+
+                var pluginPaths = AssetDatabase.FindAssets("", new[] { pluginDirectoryCompatibility.Path });
+                foreach (var pluginPath in pluginPaths)
+                {
+                    var plugin = AssetImporter.GetAtPath(AssetDatabase.GUIDToAssetPath(pluginPath)) as PluginImporter;
+                    if (plugin == null)
+                    {
+                        continue;
+                    }
+
+                    // We only update options that needs to be updated to avoid reloading plugins that have correct settings
```

Comment:
It's a shame Unity doesn't do this sort of checking already...

Reply:
Yeah, I was quite disappointed to find out it doesn't